src/core/qgsmaprendererparalleljob.cpp

```diff
@@ -270,6 +270,10 @@ void QgsMapRendererParallelJob::renderLayerStatic( LayerRenderJob &job )
     QgsDebugMsg( "Caught unhandled unknown exception" );
   }
   job.renderingTime = t.elapsed();
+  if ( job.layer )
+  {
+    job.layer->setLastRenderingTime( job.renderingTime );
```

Comment:
This needs to be synchronized with the main thread, I think this method runs on a different thread (e.g. using Qt::QueuedConnection)